nginx/docker-entrypoint.d/00-envcheck.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+set -e
+
+[ -f "/etc/nginx/conf.d/default.conf" ] && rm /etc/nginx/conf.d/default.conf
+
+[ -z "$UPSTREAM_PWA" ] && echo "UPSTREAM_PWA is not set" && exit 1
+
+[ -z "$ICM_BASE_URL" ] && echo "ICM_BASE_URL is not set. Cannot use sitemap proxy feature."
```

Comment:
I only get a clean startup of the nginx in a docker compose if I rewrite these statements as `if`..`then`..`fi`. Otherwise, the nginx exits with 1.

Reply:
I think this status comes from the shell test instructions because they are not run in an `if`.

Reply:
I understand your request only thing is, I do not see that nginx exits at all. With or without the environment variables set I have a running container.

Reply:
Well, can you rewrite the one liners into proper ifs nonetheless? :sweat_smile:

Reply:
Sure I can do that. Only wanted to know under which circumstances the container exits. Maybe for a test case later on...

Reply:
I'm running the docker compose in our project similar to the standard PWA repo one. Only UPSTREAM_PWA and OVERRIDE_IDENTITY_PROVIDERS set. I run it in WSL2.

Reply:
Yes, I'm sure, just tested it. Set OVERRIDE_IDENTITY_PROVIDERS and then you'll get 
```
intershop-pwa-nginx-1 exited with code 1
```
😄

Reply:
I included some shell tests to expose the issue. As you can see, it's working. :smile:

Reply:
Tested OK in project :+1: